### ISSUE TITLE
`get_biosamples` can be improved by avoiding the join in SciDB

### ISSUE THREAD

**Issue by ksens** [Paradigm4/revealgenomics#7]
Currently `get_biosamples()` retrieves biosample info in all user-permitted namespaces. For a user with access to both `public` and `clinical` namespaces, the queries executed are:

```sh

equi_join(BIOSAMPLE, public.BIOSAMPLE_INFO, 'left_names=biosample_id', 'right_names=biosample_id', 'left_outer=true', 'keep_dimensions=true')

equi_join(clinical.BIOSAMPLE, clinical.BIOSAMPLE_INFO, 'left_names=biosample_id', 'right_names=biosample_id', 'left_outer=true', 'keep_dimensions=true')
```

The `equi_join` is used over `cross_join` because the full left-outer join is needed. And only `equi_join` supports that right now. 

The timing is quite slow:
```R
t1 = proc.time(); xx = get_biosamples(); print(nrow(xx)); print(proc.time()-t1)
# [1] 22860
#    user  system elapsed 
#   6.748   0.257  10.975 
```

Now let's try downloading the arrays into R and then doing the merge in R:

```R
t1 = proc.time(); xx = iquery(db, "merge(BIOSAMPLE, clinical.BIOSAMPLE)", return = T); print(nrow(xx)); print(proc.time()-t1)
# [1] 22860
#    user  system elapsed 
#   0.755   0.003   1.375

t1 = proc.time(); yy = iquery(db, "merge(BIOSAMPLE_INFO, clinical.BIOSAMPLE_INFO)", return = T); print(nrow(xx)); print(proc.time()-t1)
# [1] 464412
#    user  system elapsed 
#   1.410   0.040   2.584 

t1 = proc.time(); zz = merge(xx, yy, by = "biosample_id"); print(proc.time()-t1)
#    user  system elapsed 
#   1.078   0.000   1.072 
```

Total time: `1.375+2.584+1.072=5.031`.

There is still some stuff remaining (like pivot-ing the key-value pairs to columns), but avoiding the join in SciDB seems to allow a ~2x speedup. 

The difference might be more noticeable when we need to merge data across 3 or more namespaces.

**Comment by ksens** [comment]
This download strategy can be expanded to all metadata categories. See #47.

**Comment by ksens** [comment]
Closed via PR #67
